core/app/models/spree/order.rb

```diff
@@ -27,6 +27,25 @@ def generate_number(options = {})
     alias :display_ship_total :display_shipment_total
     alias_attribute :ship_total, :shipment_total
 
+    MONEY_THRESHOLD  = 999_999
```

Comment:
Constraints are easier to raise than to introduce, I generally prefer to start with something "reasonably constraining", instead of directly going with the highest number possible. While not being aware of all consequences.

There is a lot that can be "derived" from money, refunds, promotions, taxes, summaries, reports etc. If we go with a very high amount for such a base value its easy to overflow a derived one.

The moment I realize I choose a to restrictive constraint its much easier to lift, than to back-fill into a longer running project.

//cc @dkubb You also had nice words to describe this, feel free to add your perspective.